wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,418 @@
+                            "You may NOT rejoin the guild!**")
+            e_obj.set_footer(icon_url=self.bot.guilds[0].icon_url, text=self.bot.guilds[0])
+
+            try:
+                await member.send(embed=e_obj)
+            except (discord.HTTPException, discord.Forbidden, discord.InvalidArgument):
+                logger.info('[Ban watchdog()] unable to send warning dm to banned user due to user dm settings.')
+            await member.kick(reason="Not allowed back on server.")
+
+    @commands.Cog.listener(name='on_member_ban')
+    async def intercept(self, guild: discord.Guild, member: Union[discord.User, discord.Member]):
+        """Watches for a guild ban. The guild ban is undone and the user is banned via this ban system"""
+
+        # need to read the audit log to grab mod, date, and reason
+        logger.info(f"[Ban intercept()] guild ban detected and intercepted for user='{member}'")
+        try:
+            audit_ban = await self.bot.guilds[0].audit_logs(action=discord.AuditLogAction.ban).flatten()
+        except Exception as e:
+            logger.info(f'error while fetching ban data: {e}')
+            await self.mod_channel.send(f"Encountered following error while intercepting a ban: {e}\n" +
+                                        "**Most likely need view audit log perms.**")
+            return
+        audit_ban = audit_ban[0]
+        logger.info(f"[Ban intercept()] audit log data retrieved for intercepted ban: {audit_ban}")
+
+        # name, id, mod, mod id, date, reason
+        ban = BanRecords(
+                         username=member.name + '#' + member.discriminator,
+                         user_id=member.id,
+                         mod=audit_ban.user.name + '#' + audit_ban.user.discriminator,
+                         mod_id=audit_ban.user.id,
+                         ban_date=audit_ban.created_at.timestamp(),
+                         reason=audit_ban.reason if audit_ban.reason else 'No Reason Given!'
+                         )
+
+        # update ban_list and db
+        self.ban_list.append(member.id)
+        await BanRecords.insert_record(ban)
+
+        # unban
+        await self.bot.guilds[0].unban(member)
+        logger.info(f"[Ban intercept()] ban for {ban.username} moved into db and guild ban was removed")
+
+        # report to council
+        e_obj = discord.Embed(title="Ban Hammer Deployed",
+                              colour=discord.Color.red())
+
+        e_obj.add_field(name="Banned User", value=f"**{ban.username}**", inline=True)
+        e_obj.add_field(name="Moderator", value=f"**{ban.mod}**", inline=True)
+        e_obj.add_field(name="Reason", value=f"```{ban.reason}```", inline=False)
+        e_obj.add_field(name="Notification DM", value="*NOT SENT*\nCause: NO COMMON GUILD\n", inline=False)
+        e_obj.set_footer(text="Intercepted Moderator Action")
+
+        if e_obj:
```

Comment:
not much point in this conditional. if `e_obj` returns false, it would have thrown an error much earlier up, no?